README.md
Runhui Wang
A16931624
https://dainty-narwhal-ed5261.netlify.app/

Git:https://github.com/Runhui-Wang/134bhw4
Part 1:https://dainty-narwhal-ed5261.netlify.app/nativedialogs.html
Part 2:https://dainty-narwhal-ed5261.netlify.app/customdialogs.html
Part 3:https://dainty-narwhal-ed5261.netlify.app/crud.html
Part 4:https://dainty-narwhal-ed5261.netlify.app/styledcrud.html
Part 5:https://dainty-narwhal-ed5261.netlify.app

changelog.md
I added a select menu for showing project. 
I also used twitter widget js to include a post,  which will be one of my post working after it is officially deployed. 

blog.js
var crud = new function() {
    // Examples
    this.blogs = ["Webdev", "Javascript", "DOM"];
    this.times = ["1921/23/23", "2022/98/65", "1983/85/12"];
    this.summarys = ["webdev", "javascript", "dom"];
    
    // Stores current index 
    var indexRetreive = '';
    var indexRemove = '';
    
    // Display all posts
    this.DisplayAll = function() {
        var data = '';
        
        if (this.blogs.length > 0) {
            var i;
            for (i=0; i < this.blogs.length; i++) {
                data += '<li>';
                data += this.blogs[i] + ' (' + this.times[i] + ') - summary: ' + this.summarys[i];
                data += '<button id="edit" onclick="crud.Retreive(' + i + ')"> Edit</button>';
                data += '<button id="delete" onclick="crud.Remove(' + i + ')"> Delete</button>';
                data += '</li>';
            }
        }
        else {
            data = "No blogs are added";
        }
        
        return document.getElementById('postList').innerHTML = data;
    };
    
    // Function that shows add blog dialog box
    this.Add = function() {
        document.getElementById('title').value = "";
        document.getElementById('time').value = "";
        document.getElementById('userInput').showModal();
    };
    
    // Insert new 
    this.Insert = function() {
        var title = document.getElementById('title').value;
        var time = document.getElementById('time').value;
        var summary = document.getElementById('summary').value;
        if (title || time) {
            this.blogs.push(title.trim());
            this.times.push(time.trim());
            this.summarys.push(summary.trim());
            this.DisplayAll();
        }
        document.getElementById('userInput').close();
    };

    // Cancel blog input
    this.cancelAdd = function() {
        document.getElementById('userInput').close();    
    };
    
    // Opens edit dialog box
    this.Retreive = function(item) {
        indexRetreive = item;
        document.getElementById('titleEdit').value = this.blogs[item];
        document.getElementById('timeEdit').value = this.times[item];
        document.getElementById('summaryEdit').value = this.summarys[item];
        document.getElementById('userInput2').showModal();
    };
    
    // Edit selected blog that is already in array
    this.Edit = function() {
        var title = document.getElementById('titleEdit').value;
        var time = document.getElementById('timeEdit').value;
        var summary = document.getElementById('summaryEdit').value;
        var self = this;      
        if (title || time) {
            self.blogs.splice(indexRetreive, 1, title.trim());
            self.times.splice(indexRetreive, 1, time.trim());
            self.summarys.splice(indexRetreive, 1, summary.trim());
            self.DisplayAll();
        }
        document.getElementById('userInput2').close();
    };
    
    // Cancel blog edit
    this.cancelEdit = function() {
        document.getElementById('userInput2').close();    
    };
 
    // Opens delete dialog box
    this.Remove = function(item) {
        indexRemove = item;
        document.getElementById('deleteblog').showModal();
    }; 
    
    // Delete blog from arrays
    this.Delete = function() {
        this.blogs.splice(indexRemove, 1);
        this.times.splice(indexRemove, 1);
        this.summarys.splice(indexRemove, 1);
        this.DisplayAll();
        document.getElementById('deleteblog').close();        
    };
   
    // Cancel delete
    this.cancelDelete = function() {
        document.getElementById('deleteblog').close();    
    };    
}

crud.DisplayAll();

portfolio.js
function read_projects(){
    let select_value = document.getElementById("pjs").value;
    if(select_value == "c++ project"){
        document.getElementById("c++").style.display = "initial";
        document.getElementById("c").style.display = "none";
        document.getElementById("python").style.display = "none";
        document.getElementById("matlab").style.display = "none";
    }
    else if(select_value == "c project"){
        document.getElementById("c++").style.display = "none";
        document.getElementById("c").style.display = "initial";
        document.getElementById("python").style.display = "none";
        document.getElementById("matlab").style.display = "none";
    }
    else if(select_value == "Python project"){
        document.getElementById("c++").style.display = "none";
        document.getElementById("c").style.display = "none";
        document.getElementById("python").style.display = "initial";
        document.getElementById("matlab").style.display = "none";
    }

    else if(select_value == "MatLab project"){
        document.getElementById("c++").style.display = "none";
        document.getElementById("c").style.display = "none";
        document.getElementById("python").style.display = "none";
        document.getElementById("matlab").style.display = "initial";
    }
    else{
        document.getElementById("c++").style.display = "none";
        document.getElementById("c").style.display = "none";
        document.getElementById("python").style.display = "none";
        document.getElementById("matlab").style.display = "none";
    }
}
function handleClick() {
    read_projects();
}


customdialog.js
const alertBtn = document.getElementById('alert');
const confirmBtn = document.getElementById('confirm');
const promptBtn = document.getElementById('prompt');
const output = document.querySelector('output');
const alertDialog = document.getElementById('alertDialog');
const confirmDialog = document.getElementById('confirmDialog');
const promptDialog = document.getElementById('promptDialog');

alertBtn.addEventListener('click', () => {
        output.innerHTML = "";

        alertDialog.showModal();

})
confirmBtn.addEventListener('click', () => {
   
        output.innerHTML = "";

        confirmDialog.showModal();
  

})
promptBtn.addEventListener('click', () => {
    if (typeof alertDialog.showModal === "function"){
        const output = document.querySelector('output');
        output.innerHTML = "";

        promptDialog.showModal();
    }
    else{
        alert("this browser does not support <dialog>")
    }
})

function alertchange(dialog){

    dialog.addEventListener('close', () => {
        const output = document.querySelector('output');
        output.innerHTML = "You clicked an alert!!!";
        
    })

}
function confirmchange(dialog){

    dialog.addEventListener('close', () => {
        const answer = dialog.returnValue;
        if(answer == "1"){
            output.innerHTML = "The value returned by the confirm method is : True";
            
        }
        else{
            output.innerHTML = "The value returned by the confirm method is : False";

        }
        
    })

}

function promptchange(dialog){

    dialog.addEventListener('close', () => {
        const text = document.querySelector('input')
        let feedback = promptDialog.returnValue
        let clean = "a";
        

        if(feedback == "clprompt"){
            output.innerHTML = "User didn't enter anything";
            
        }
        else{
            clean = DOMPurify.sanitize(text.value);
            output.innerHTML = `The value returned by the prompt method is : ${clean}`;

        }
    })

}

alertchange(alertDialog);
confirmchange(confirmDialog);
promptchange(promptDialog);

styledposts.js
var crud = new function() {
    // Examples
    this.blogs = ["Webdev", "Javascript", "DOM"];
    this.times = ["1921/23/23", "2022/98/65", "1983/85/12"];
    this.summarys = ["webdev", "javascript", "dom"];
    
    // Stores current index 
    var indexRetreive = '';
    var indexRemove = '';
    
    // Display all posts
    this.DisplayAll = function() {
        var data = '';
        
        if (this.blogs.length > 0) {
            var i;
            for (i=0; i < this.blogs.length; i++) {
                //data += '<li>';
                data += '<b>Title</b>: '+this.blogs[i] 
                data += '<br>'
                data += ' <b>Date</b>:' + this.times[i] 
                data += '<br>'
                data += ' <b>Summary</b>: ' + this.summarys[i];
                data += '<br>'
                data += '<button id="edit" onclick="crud.Retreive(' + i + ')">✏️ Edit</button>';
                data += '<button id="delete" onclick="crud.Remove(' + i + ')">🗑️Delete</button>' + '<br>';
                //data += '</li>';
            }
        }
        else {
            data = "No blogs are added";
        }
        
        return document.getElementById('postList').innerHTML = data;
    };
    
    // Function that shows add blog dialog box
    this.Add = function() {
        document.getElementById('title').value = "";
        document.getElementById('time').value = "";
        document.getElementById('userInput').showModal();
    };
    
    // Insert new 
    this.Insert = function() {
        var title = document.getElementById('title').value;
        var time = document.getElementById('time').value;
        var summary = document.getElementById('summary').value;
        if (title || time) {
            this.blogs.push(title.trim());
            this.times.push(time.trim());
            this.summarys.push(summary.trim());
            this.DisplayAll();
        }
        document.getElementById('userInput').close();
    };

    // Cancel blog input
    this.cancelAdd = function() {
        document.getElementById('userInput').close();    
    };
    
    // Opens edit dialog box
    this.Retreive = function(item) {
        indexRetreive = item;
        document.getElementById('titleEdit').value = this.blogs[item];
        document.getElementById('timeEdit').value = this.times[item];
        document.getElementById('summaryEdit').value = this.summarys[item];
        document.getElementById('userInput2').showModal();
    };
    
    // Edit selected blog that is already in array
    this.Edit = function() {
        var title = document.getElementById('titleEdit').value;
        var time = document.getElementById('timeEdit').value;
        var summary = document.getElementById('summaryEdit').value;
        var self = this;      
        if (title || time) {
            self.blogs.splice(indexRetreive, 1, title.trim());
            self.times.splice(indexRetreive, 1, time.trim());
            self.summarys.splice(indexRetreive, 1, summary.trim());
            self.DisplayAll();
        }
        document.getElementById('userInput2').close();
    };
    
    // Cancel blog edit
    this.cancelEdit = function() {
        document.getElementById('userInput2').close();    
    };
 
    // Opens delete dialog box
    this.Remove = function(item) {
        indexRemove = item;
        document.getElementById('deleteblog').showModal();
    }; 
    
    // Delete blog from arrays
    this.Delete = function() {
        this.blogs.splice(indexRemove, 1);
        this.times.splice(indexRemove, 1);
        this.summarys.splice(indexRemove, 1);
        this.DisplayAll();
        document.getElementById('deleteblog').close();        
    };
   
    // Cancel delete
    this.cancelDelete = function() {
        document.getElementById('deleteblog').close();    
    };    
}

crud.DisplayAll();